### PR TITLE
Firebase Storage Implementation with functional Profile Picture Upload

### PR DESCRIPTION
**Pull request by achillet3**
# Implementation of Firebase Storage
- Addition of functions to upload an to the Firebase Storage and retrieve that image 
- Upon sign up upload the chosen picture to the Firebase storage and set the profilePictureUrl document of the user to the link of that image in the database
    - If the user does not choose a picture it will automatically be set the place holder image form the Firebase storage

### CODE REVIEW

File: app/src/main/java/com/github/se/eventradar/viewmodel/AuthenticationViewModel.kt [hunk 1]
```diff
@@ -70,6 +88,38 @@ class LoginViewModel @Inject constructor(private val userRepository: IUserReposi
...
   }
 
+  private suspend fun uploadImageAsync(
+      selectedImageUri: Uri?,
+      uid: String,
+      folderName: String
+  ): Boolean {
+    if (selectedImageUri == null) {
+      return false
+    }
```
Comment (Viccha18) [inline app/src/main/java/com/github/se/eventradar/viewmodel/AuthenticationViewModel.kt:98]: Where do you set it to be the placeholder when the user don't select an image ?

File: app/src/main/res/values/strings.xml [hunk 2]
```diff
@@ -24,6 +24,8 @@
     <string name="homeScreen_events">Home</string>
     <string name="user_profile">Profile</string>
     <string name="my_hosted_events">My hosting</string>
+    <string name="profile_pics_folder">Profile_Pictures</string>
+    <string name="qr_codes_folder">QR_Codes</string>
```
Comment (Viccha18) [inline app/src/main/res/values/strings.xml:28]: Not sure where you use these Resource String

**Review by Viccha18 — APPROVED** [review]
LGTM, I only have a few questions

File: app/src/main/java/com/github/se/eventradar/model/repository/user/FirebaseUserRepository.kt [hunk 3]
```diff
@@ -136,6 +138,30 @@ class FirebaseUserRepository(db: FirebaseFirestore = Firebase.firestore) : IUser
...
     }
   }
+
+  override suspend fun uploadImage(
+      selectedImageUri: Uri,
+      uid: String,
+      folderName: String
+  ): Resource<Unit> {
+    val storageRef = Firebase.storage.reference.child("$folderName/$uid")
+    return try {
```
Comment (wisdomseeker2001) [inline app/src/main/java/com/github/se/eventradar/model/repository/user/FirebaseUserRepository.kt:148]: will ur try block ever fail with this struct?

**Review by wisdomseeker2001 — COMMENTED** [review]
Looks good to me on a global basis

**Comment by jgeaso1266** [comment]
resolver by #176

**Pull request closed without merging**